packages/discord_gcp_alerts/lib/src/models/gcp_alert.dart

```diff
@@ -0,0 +1,244 @@
+
+  /// The type of the resource.
+  final String? type;
+}
+
+/// {@template metric}
+/// A metric associated with an incident.
+/// {@endtemplate}
+@JsonSerializable()
+class Metric {
+  /// {@macro metric}
+  const Metric({this.type, this.displayName});
+
+  /// Creates a [Metric] from a JSON [Map].
+  factory Metric.fromJson(Map<String, dynamic> json) => _$MetricFromJson(json);
+
+  /// Converts a [Metric] to the JSON representation.
+  Map<String, dynamic> toJson() => _$MetricToJson(this);
+
+  /// The type of the metric.
+  final String? type;
+
+  /// The display name of the metric.
+  final String? displayName;
+}
+
+/// {@template condition}
+/// A condition associated with an incident.
+/// {@endtemplate}
+@JsonSerializable()
+class Condition {
+  /// {@macro condition}
+  const Condition({this.name, this.displayName, this.conditionThreshold});
+
+  /// Creates a [Condition] from a JSON [Map].
+  factory Condition.fromJson(Map<String, dynamic> json) =>
+      _$ConditionFromJson(json);
+
+  /// Converts a [Condition] to the JSON representation.
+  Map<String, dynamic> toJson() => _$ConditionToJson(this);
+
+  /// The name of the condition.
+  final String? name;
+
+  /// The display name of the condition.
+  final String? displayName;
+
+  /// The condition threshold.
+  final ConditionThreshold? conditionThreshold;
+}
+
+/// {@template condition_threshold}
+/// The condition threshold associated with an incident.
+/// {@endtemplate}
+@JsonSerializable()
+class ConditionThreshold {
+  /// {@macro condition_threshold}
+  const ConditionThreshold({
+    this.filter,
+    this.comparison,
+    this.thresholdValue,
+    this.duration,
+    this.trigger,
+  });
+
+  /// Creates a [ConditionThreshold] from a JSON [Map].
+  factory ConditionThreshold.fromJson(Map<String, dynamic> json) =>
+      _$ConditionThresholdFromJson(json);
+
+  /// Converts a [ConditionThreshold] to the JSON representation.
+  Map<String, dynamic> toJson() => _$ConditionThresholdToJson(this);
+
+  /// The filter associated with the condition threshold.
+  final String? filter;
+
+  /// The comparison associated with the condition threshold.
+  final String? comparison;
+
+  /// The threshold value associated with the condition threshold.
+  final double? thresholdValue;
+
+  /// The duration associated with the condition threshold.
+  final String? duration;
+
+  /// The trigger associated with the condition threshold.
+  final Trigger? trigger;
+}
+
+/// {@template trigger}
+/// The trigger associated with an incident.
+/// {@endtemplate}
+@JsonSerializable()
+class Trigger {
+  /// {@macro trigger}
+  const Trigger({this.count});
+
+  /// Creates a [Trigger] from a JSON [Map].
+  factory Trigger.fromJson(Map<String, dynamic> json) =>
+      _$TriggerFromJson(json);
+
+  /// Converts a [Trigger] to the JSON representation.
+  Map<String, dynamic> toJson() => _$TriggerToJson(this);
+
+  /// The count associated with the trigger.
+  final int? count;
```

Comment:
I'm a little confused by this object. What does `Trigger` mean? I'd expect it to be a richer object with a description etc, but the fact that it only has a count property suggests otherwise.

Reply:
See https://cloud.google.com/monitoring/alerts/policies-in-json

Reply:
Good lord these docs are awful